geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/utils/HBaseBatchScan.scala

```diff
@@ -23,13 +23,14 @@ import org.opengis.filter.Filter
 private class HBaseBatchScan(connection: Connection, table: TableName, ranges: Seq[Scan], threads: Int, buffer: Int)
     extends AbstractBatchScan[Scan, Result](ranges, threads, buffer, HBaseBatchScan.Sentinel) {
 
+
   private val htable = connection.getTable(table, new CachedThreadPool(threads))
 
   override protected def scan(range: Scan, out: BlockingQueue[Result]): Unit = {
     val scan = htable.getScanner(range)
     try {
       var result = scan.next()
-      while (result != null) {
+      while (result != null && !closed) {
         out.put(result)
         result = scan.next()
```

Comment:
maybe we should switch this around slightly to avoid pulling back a result we don't read:
```
var result = scan.next() // first call we shouldn't need to check since it's done in SingleThreadScan
while (result != null) {
  out.put(result)
  result = if (closed) { null } else { scan.next() }
}
```

Reply:
Makes sense.  Pushing this change!

Reply:
Resolving this conversation since the build passed!